### ISSUE TITLE
Certain order of operations leads to Error thrown on #set

### ISSUE THREAD

**Issue by hagemt** [avoidwork/tiny-lru#4]
Reproduction steps; happened in our use of `_.memoize` where `cache = LRU()`:

```javascript
> const cache = require('tiny-lru')(10)
undefined
> cache.set('1', 'a')
LRU {
  cache: { '1': { next: null, previous: null, value: 'a' } },
  first: '1',
  last: '1',
  length: 1,
  max: 10,
  notify: false,
  onchange: [Function],
  update: [Function] }
> cache.set('2', 'b')
LRU {
  cache: 
   { '1': { next: '2', previous: null, value: 'a' },
     '2': { next: null, previous: '1', value: 'b' } },
  first: '2',
  last: '1',
  length: 2,
  max: 10,
  notify: false,
  onchange: [Function],
  update: [Function] }
> cache.set('1', 'c')
LRU {
  cache: 
   { '1': { next: null, previous: '2', value: 'c' },
     '2': { next: '1', previous: '1', value: 'b' } },
  first: '1',
  last: '1',
  length: 2,
  max: 10,
  notify: false,
  onchange: [Function],
  update: [Function] }
> cache.delete(1)
{ next: null, previous: '2', value: 'c' }
> cache
LRU {
  cache: { '2': { next: null, previous: '1', value: 'b' } },
  first: '1',
  last: '1',
  length: 1,
  max: 10,
  notify: false,
  onchange: [Function],
  update: [Function] }
> cache.set('3', 'd')
TypeError: Cannot set property 'next' of undefined
    at LRU.set (/home/twi/Code/CareAssistant/node_modules/tiny-lru/lib/tiny-lru.es6.js:141:33)
    at repl:1:7
    at ContextifyScript.Script.runInThisContext (vm.js:23:33)
    at REPLServer.defaultEval (repl.js:340:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.onLine (repl.js:537:10)
    at emitOne (events.js:101:20)
    at REPLServer.emit (events.js:189:7)
    at REPLServer.Interface._onLine (readline.js:238:10)
```

**Comment by avoidwork** [comment]
@hagemt what version are you using? line 141 is now...

`if (this.length === 1) {`

**Comment by avoidwork** [comment]
still happening in latest build, now at line 153; thanks for the steps to reproduce, i'll get a fix out later today!

**Comment by avoidwork** [comment]
@hagemt the fix will land in 1.4.8, please update your package.json; the difference between 1.3 & 1.4 is optional sync code paths for multi-service normalization.

**Comment by avoidwork** [comment]
it's on npm now 👍
